share-extension/ShareViewController.m

```diff
@@ -1156,4 +1157,19 @@ - (UIImage *) rotateImage:(UIImage *)image {
     return newImage;
 }
 
+-(NSString *) MIMETypeForFile:(NSString *)fileName {
+    NSRange lastPointRange = [fileName rangeOfString:@"." options:NSBackwardsSearch];
+    if (lastPointRange.location != NSNotFound) {
+        NSString * fileExtension = [fileName substringFromIndex:lastPointRange.location+1];
+        CFStringRef pathExtension = (__bridge CFStringRef)(fileExtension);
+        CFStringRef type = UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, pathExtension, NULL);
+        CFRelease(pathExtension);
+        // The UTI can be converted to a mime type:
+        NSString *mimeType = (__bridge_transfer NSString *)UTTypeCopyPreferredTagWithClass(type, kUTTagClassMIMEType);
+        return mimeType;
+    }
+    
+    return @"text/html";
```

Comment:
Is there any reason why you're forcing the mime type to `text/html` if the real type can't be found ?
If not, then you should return `nil` and not add the `mimeType` property in the template params.